test/test_transforms.py

```diff
@@ -1403,6 +1322,88 @@ def test_random_erasing(self):
         t.__repr__()
 
 
+@pytest.mark.parametrize('channels', [1, 3, 4])
+def test_pil_to_tensor(channels):
+    height, width = 4, 4
+    trans = transforms.PILToTensor()
+
+    with pytest.raises(TypeError):
+        trans(np.random.rand(1, height, width).tolist())
+        trans(np.random.rand(1, height, width))
+
+    input_data = torch.ByteTensor(channels, height, width).random_(0, 255)
+    img = transforms.ToPILImage()(input_data)
+    output = trans(img)
+    torch.testing.assert_close(input_data, output, check_stride=False)
+
+    input_data = np.random.randint(low=0, high=255, size=(height, width, channels)).astype(np.uint8)
+    img = transforms.ToPILImage()(input_data)
+    output = trans(img)
+    expected_output = input_data.transpose((2, 0, 1))
+    torch.testing.assert_close(output.numpy(), expected_output)
+
+    input_data = torch.as_tensor(np.random.rand(channels, height, width).astype(np.float32))
+    img = transforms.ToPILImage()(input_data)  # CHW -> HWC and (* 255).byte()
+    output = trans(img)  # HWC -> CHW
+    expected_output = (input_data * 255).byte()
+    torch.testing.assert_close(output, expected_output, check_stride=False)
+
+    # separate test for mode '1' PIL images
+    input_data = torch.ByteTensor(1, height, width).bernoulli_()
+    img = transforms.ToPILImage()(input_data.mul(255)).convert('1')
+    output = trans(img).view(torch.uint8).bool().to(torch.uint8)
+    torch.testing.assert_close(input_data, output, check_stride=False)
+
+
+@pytest.mark.parametrize('channels', [1, 3, 4])
+def test_to_tensor(channels):
+    height, width = 4, 4
+    trans = transforms.ToTensor()
+
+    with pytest.raises(TypeError):
+        trans(np.random.rand(1, height, width).tolist())
+
+    with pytest.raises(ValueError):
+        trans(np.random.rand(height))
+        trans(np.random.rand(1, 1, height, width))
```

Comment:
same here about having 1 `raises` block per line, and we should also move these out in a separate test function